lib/backend/k8s/resources/customresource.go

```diff
@@ -363,10 +299,31 @@ func (c *customK8sResourceClient) List(ctx context.Context, list model.ListInter
 	}
 	return &model.KVPairList{
 		KVPairs:  kvps,
-		Revision: revision,
+		Revision: reslOut.GetListMeta().GetResourceVersion(),
 	}, nil
 }
 
+func (c *customK8sResourceClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
+	resl, ok := list.(model.ResourceListOptions)
+	if !ok {
+		return nil, errors.New("internal error: custom resource watch invoked for non v2 resource type")
+	}
+	if len(resl.Name) != 0 {
+		return nil, fmt.Errorf("cannot watch specific resource instance: %s", resl.Name)
+	}
+
+	k8sWatchClient := cache.NewListWatchFromClient(
+		c.restClient,
+		c.resource,
+		resl.Namespace,
+		fields.Everything())
+	k8sWatch, err := k8sWatchClient.WatchFunc(metav1.ListOptions{ResourceVersion: revision})
+	if err != nil {
+		return nil, err
```

Comment:
Do we need to convert this error to a Calico error?

Reply:
I think we should, yes!